openmmtools/integrators.py

```diff
@@ -1185,25 +1095,562 @@ def substep_function(step_string):
+
+    def substep_function(self, step_string, measure_shadow_work, measure_heat, n_R, force_group_nV, mts):
+        """
+        Take step string, and add the appropriate R, V, O, M, or H step with appropriate parameters.
+        The step string input here is a single character (or character + number, for MTS)
+
+        Parameters
+        ----------
+        step_string : str
+            R, O, V, or Vn (where n is a nonnegative integer specifying force group)
+        measure_shadow_work : bool
+            Whether the steps should measure shadow work
+        measure_heat : bool
+            Whether the O step should measure heat
+        n_R : int
+            The number of R steps per integrator step
+        force_group_nV : dict
+            The number of V steps per integrator step per force group. {0: nV} if not mts
+        mts : bool
+            Whether the integrator is a multiple timestep integrator
+        """
+
+        if step_string == "O":
+            self.O_step(measure_heat)
+        elif step_string == "R":
+            self.R_step(measure_shadow_work, n_R)
+        elif step_string == "{":
+            self.addComputePerDof("xold", "x")
+            self.addComputePerDof("vold", "v")
+        elif step_string == "}":
+            self.metropolize()
+        elif step_string[0] == "V":
+            #get the force group for this update--it's the number after the V
+            force_group = step_string[1:]
+            self.V_step(force_group, measure_shadow_work, force_group_nV, mts)
+        elif step_string == "H":
+            self.addAlchemicalPerturbationStep()
+
+    def addGlobalVariables(self, nsteps):
+        """
+        Add the appropriate global parameters to the CustomIntegrator. nsteps refers to the number of
+        total steps in the protocol.
+
+        Parameters
+        ----------
+        nsteps : int, greater than 0
+            The number of steps in the switching protocol.
+        """
+        self.addGlobalVariable('lambda', 0.0) # parameter switched from 0 <--> 1 during course of integrating internal 'nsteps' of dynamics
+        self.addGlobalVariable('kinetic', 0.0) # kinetic energy
+        self.addGlobalVariable('nsteps', nsteps) # total number of NCMC steps to perform
+        self.addGlobalVariable('step', 0) # current NCMC step number
+
+class ExternalPerturbationLangevinSplittingIntegrator(LangevinSplittingIntegrator):
+    """
+    LangevinSplittingIntegrator that accounts for external perturbations and tracks protocol work.
+    """
+
+    def __init__(self,
+                 splitting="V R O R V",
+                 temperature=298.0 * simtk.unit.kelvin,
+                 collision_rate=1.0 / simtk.unit.picoseconds,
+                 timestep=1.0 * simtk.unit.femtoseconds,
+                 constraint_tolerance=1e-8,
+                 measure_shadow_work=False,
+                 measure_heat=True):
+
+        self.addGlobalVariable("protocol_work", 0)
```

Comment:
We can't define global variables before calling the CustomIntegrator init function. 

I'm running into this error:
```pytb
File "C:\GitHub\openmmtools\openmmtools\tests\test_integrators.py", line 219, in test_vvvr_protocol_work_accumulation
    integrator = integrators.ExternalPerturbationLangevinSplittingIntegrator(splitting="O V R V O", temperature=temperature)
  File "C:\GitHub\openmmtools\openmmtools\integrators.py", line 1638, in __init__
    self.addGlobalVariable("protocol_work", 0)
  File "C:\Users\rustenba\AppData\Local\Continuum\Miniconda3\envs\py35\lib\site-packages\simtk\openmm\openmm.py", line 10574, in addGlobalVariable
    return _openmm.CustomIntegrator_addGlobalVariable(self, name, initialValue)
Exception: in method 'CustomIntegrator_addGlobalVariable', argument 1 of type 'OpenMM::CustomIntegrator *'
```